tests/observability/network/conftest.py

```diff
@@ -191,3 +194,23 @@ def invalid_cnao_operator(prometheus, admin_client, hco_namespace, csv_scope_cla
 
     linux_bridge_plugin_ds = DaemonSet(name=KUBE_CNI_LINUX_BRIDGE_PLUGIN, namespace=hco_namespace.name)
     linux_bridge_plugin_ds.wait_until_deployed(timeout=TIMEOUT_10MIN)
+
+
+@pytest.fixture(scope="class")
+def wait_csv_image_updated_with_bad_image(csv_with_invalid_cnao_operator):
+    def _get_csv_image_from_cnao():
+        for deployment in csv_with_invalid_cnao_operator.instance.spec.install.spec.deployments:
+            if deployment.name == CLUSTER_NETWORK_ADDONS_OPERATOR:
+                return deployment.spec.template.spec.containers[0].image
+
+    sample = TimeoutSampler(
+        wait_timeout=TIMEOUT_4MIN,
+        sleep=TIMEOUT_10SEC,
+        func=_get_csv_image_from_cnao,
+    )
+    try:
+        for sample in sample:
+            if sample == NON_EXISTS_IMAGE:
+                return
+    except TimeoutExpiredError:
+        raise
```

Comment:
```suggestion
    for sample in TimeoutSampler(
        wait_timeout=TIMEOUT_4MIN,
        sleep=TIMEOUT_10SEC,
        func=_get_csv_image_from_cnao,
    ):
            if sample == NON_EXISTS_IMAGE:
                return
```

raise is not needed; will be handled by TimeoutSampler